### DISCUSSION TITLE
关于tree的add回调操作问个问题，这个key值是干嘛的？

### DISCUSSION THREAD

**Discussion by boshaoting** [layui/layui#1867] [问答]
我看官方文档是这样的，但是ctrl+f也没有关于key的更多信息了。：
operate: function(obj){
  var type = obj.type; // 得到操作类型：add、edit、del
  var data = obj.data; // 得到当前节点的数据
  var elem = obj.elem; // 得到当前节点元素
  
  // Ajax 操作
  var id = data.id; // 得到节点索引
  if(type === 'add'){ // 增加节点
    //返回 key 值
    return 123;

主要是我现在遇到这么个问题，click什么的函数都是通过obj.data.id获取id再去调用接口，但是在add代码里头的，obj指的是添加了新子节点的那个，也就是父节点，add出来的这个新节点就不知道怎么才能把id绑上去了。

**Comment by bxjt123** [comment]
其实试一下就知道了，`add`里`return`的这个`key`就是新节点的`id`

**Reply by boshaoting** [reply]
试过了，不行， console.log(obj.data.id); console.log(obj.id);这俩的结果都是undefined。那有可能是个bug

**Reply by bxjt123** [reply]
我试的三个回调函数均可得到新节点自定义的`id`
![111](https://github.com/layui/layui/assets/33601030/03c33715-b765-4a22-887e-ff0084d80953)

**Reply by boshaoting** [reply]
非常感谢大佬愿意测试，是我代码的问题，我把return id写在ajax里导致的，那怎么把获取的res.data.id绑定到新节点呢，在ajax的代码块里又没法操作新节点。看来只能改用同步请求了。

**Reply by bxjt123** [reply]
我自己是新建节点后直接重载数据，页面数据如果不多重载页面也可以
